src/taskgraph/transforms/fetch.py

```diff
@@ -9,91 +9,76 @@
-                """.lstrip()
-            ),
-        ): str,
-        Optional(
-            "task-from",
-            description=dedent(
-                """
-                Relative path (from config.path) to the file the task was defined
-                in.
-                """.lstrip()
-            ),
-        ): str,
-        Required(
-            "description",
-            description=dedent(
-                """
-                Description of the task.
-                """.lstrip()
-            ),
-        ): str,
-        Optional("expires-after"): str,
-        Optional("docker-image"): object,
-        Optional(
-            "fetch-alias",
-            description=dedent(
-                """
-                An alias that can be used instead of the real fetch task name in
-                fetch stanzas for tasks.
-                """.lstrip()
-            ),
-        ): str,
-        Optional(
-            "artifact-prefix",
-            description=dedent(
-                """
-                The prefix of the taskcluster artifact being uploaded.
-                Defaults to `public/`; if it starts with something other than
-                `public/` the artifact will require scopes to access.
-                """.lstrip()
-            ),
-        ): str,
-        Optional("attributes"): {str: object},
-        Required("fetch"): {
-            Required("type"): str,
-            Extra: object,
-        },
-    }
-)
+class FetchConfig(Schema, rename=None, omit_defaults=False):
+    """Configuration for a fetch task type."""
+
+    type: str
+    # Additional fields handled dynamically by fetch builders
+
+
+class FetchSchema(Schema):
+    # Required fields
+    # Name of the task.
+    name: str
+    # Description of the task.
+    description: str
+    # Fetch configuration with type and additional fields.
+    fetch: Dict[str, Any]  # Must have 'type' key, other keys depend on type
+
+    # Optional fields
+    # Relative path (from config.path) to the file the task was defined in.
+    task_from: Optional[str] = None
+    # When the task expires.
+    expires_after: Optional[str] = None
+    # Docker image configuration.
+    docker_image: Optional[Any] = None
+    # An alias that can be used instead of the real fetch task name in
+    # fetch stanzas for tasks.
+    fetch_alias: Optional[str] = None
+    # The prefix of the taskcluster artifact being uploaded.
+    # Defaults to `public/`; if it starts with something other than
+    # `public/` the artifact will require scopes to access.
+    artifact_prefix: Optional[str] = None
+    # Task attributes.
+    attributes: Optional[Dict[str, Any]] = None
+
+    def __post_init__(self):
+        # Validate that fetch has a 'type' field
+        if not isinstance(self.fetch, dict) or "type" not in self.fetch:
+            raise msgspec.ValidationError("fetch must be a dict with a 'type' field")
+
+
+# Backward compatibility
+FETCH_SCHEMA = FetchSchema
```

Comment:
We're already breaking backwards compatibility, so I don't think we should worry about preserving the old variable names. I'll be easy enough to update.